src/Client.js

```diff
@@ -969,7 +1025,88 @@ class Client extends EventEmitter {
             return await window.WWebJS.getChat(chatId);
         }, chatId);
 
-        return ChatFactory.create(this, chat);
+        return chat
+            ? ChatFactory.create(this, chat)
+            : chat;
+    }
+
+    /**
+     * @typedef {Object} GetChannelOptions
+     * @property {boolean} [getMetadata = false] If true, gets the {@link ChannelMetadata}, false by default
+     */
+
+    /**
+     * @typedef {Object} ChannelMetadata
+     * @property {string} id The channel ID in a format: 'XXXXXXXXXX@newsletter'
+     * @property {number} createdAtTs The timestamp the channel was created at
+     * @property {Object} titleMetadata
+     * @property {string} titleMetadata.title The channel title
+     * @property {number} titleMetadata.updatedAtTs The timestamp the title was updated at
+     * @property {Object} descriptionMetadata
+     * @property {string} descriptionMetadata.description The channel description
+     * @property {string} descriptionMetadata.updatedAtTs The timestamp the description was updated at
+     * @property {string} inviteLink The channel invite link in a format: https://whatsapp.com/channel/INVITE_CODE
+     * @property {string} membershipType The membership type of a current user in a channel (viewer/guest/subscriber/admin/owner)
+     * @property {string} stateType The channel state type (active/suspended/geosuspended)
+     * @property {string} pictureUrl The channel profile picture url if set
+     * @property {number} subscribersCount The number of channel subscribers
+     * @property {boolean} isVerified Indicates if a channel is verified
+     */
+
+    /**
+     * Gets a {@link Channel} object or a {@link ChannelMetadata} by its ID
+     * @param {string} channelId 
+     * @param {GetChannelOptions} options
+     * @returns {Promise<ChannelMetadata|Channel>}
+     */
+    async getChannelById(channelId, options = {}) {
+        const result = await this.pupPage.evaluate(async (channelId, options) => {
+            if (options.getMetadata) {
+                try {
+                    return await window.WWebJS.getChannelMetadata(channelId, { getById: true });
+                } catch (err) {
+                    if (err.name === 'ServerStatusCodeError') return null;
+                    throw err;
+                }
+            }
+            return await window.WWebJS.getChat(channelId);
+        }, channelId, options);
+
+        return result ? (options.getMetadata ? result : ChatFactory.create(this, result)) : result;
```

Comment:
If result is undefined, its better to write that we retuening undefined

Reply:
@shirser121 
> If result is undefined, its better to write that we retuening undefined

[Fixed](https://github.com/pedroslopez/whatsapp-web.js/pull/2620/commits/8ed8626b268e99b2b5ade2b46b654a33334811cf), now returns `undefined` in a case if a `result` is falsy